internal/kafka/test/integration/kafkas_test.go

```diff
@@ -2396,6 +2397,93 @@ func TestKafka_RemovingExpiredKafkas_NoStandardInstances(t *testing.T) {
 	Expect(kafkaDeletionErr).NotTo(HaveOccurred(), "Error waiting for kafka deletion: %v", kafkaDeletionErr)
 }
 
+func TestKafka_ReassignKafkaClusterFull(t *testing.T) {
+	testServer := setup(t, func(account *v1.Account, cid string, h *coreTest.Helper) jwt.MapClaims {
+		username, _ := account.GetUsername()
+		return jwt.MapClaims{
+			"username": username,
+			"iss":      test.TestServices.KeycloakConfig.SSOProviderRealm().ValidIssuerURI,
+			"realm_access": map[string][]string{
+				"roles": {"kas_fleetshard_operator"},
+			},
+			"clientId": fmt.Sprintf("kas-fleetshard-agent-%s", cid),
+		}
+	}, nil)
+	defer testServer.TearDown()
```

Comment:
I think [t.CleanUp](https://pkg.go.dev/testing#T.Cleanup) is better suited for these kind of clean ups. WDYT?

Reply:
I was wondering why we do not use `t.CleanUp` at all in the integration, do you @ziccardi? I am thinking, the teardown method can be autowired in the cleanup function as part of the test setup as well.

Reply:
@machi1990  Yeah, I had the same thought. I think we should create a task on evaluating that option.